timesheet_activity_report/report/timesheet_activity_report.py

```diff
@@ -2,125 +2,120 @@
 # © 2015 Elico corp (www.elico-corp.com)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp.osv import fields, osv
-from openerp import tools
+from openerp import models, fields, tools
 
 
-class TimesheetReport(osv.Model):
-    """ Timesheet Activities Report """
+class TimesheetReport(models.Model):
+    """
+        Timesheet Activities Report.
+    """
     _name = "timesheet.activity.report"
+    # Database table should not be created, use init() instead
     _auto = False
     _description = "Timesheet Activities Report"
+    # Field used for the Name
     _rec_name = 'activity_name'
 
-    _columns = {
-        'id': fields.integer('Id', readonly=True),
-        'activity_type': fields.selection(
-            [
-                ('task', 'Task'),
-                ('issue', 'Issue'),
-                ('timesheet', 'Timesheet'),
-            ], 'Type', readonly=True,
-            help="Type is used to separate Tasks, Issues and\
-                Timesheets input directly"),
-        'description': fields.char('Description', readonly=True),
-        'hours': fields.float(
-            'Time spent', digits=(16, 2), readonly=True,
-            help="Time spent on timesheet"),
-        'user_id': fields.many2one('res.users', 'User', readonly=True),
-        'product_id': fields.many2one(
-            'product.product', 'Product', readonly=True),
-        'date': fields.date('Date', readonly=True),
-        'project_id': fields.many2one(
-            'project.project', 'Project', readonly=True),
-        'project_state': fields.char(
-            'State', readonly=True, help="Project State"),
-        'activity_stage_id': fields.many2one(
-            'project.task.type', 'Stage',
-            readonly=True, help="Activity Stage"),
-        'account_id': fields.many2one(
-            'account.analytic.account', 'Analytic account', readonly=True),
-        'activity_id': fields.char(
-            'Activity id', readonly=True, help="Task id or Issue id"),
-        'activity_name': fields.char(
-            'Activity name', readonly=True, help="Task name or Issue name"),
-        'br_id': fields.many2one(
-            'business.requirement', 'Bus. requ.',
-            readonly=True, help="Business requirement"),
-        'partner_id': fields.many2one(
-            'res.partner', 'Customer', readonly=True),
-        'project_categ_id': fields.many2one(
-            'project.project.category',
-            'Project Cat.', readonly=True, help="Project Category"),
-    }
+    id = fields.Integer('Id', readonly=True)
+    activity_type = fields.Selection(
+        [
+            ('task', 'Task'),
+            ('issue', 'Issue'),
+            ('timesheet', 'Timesheet'),
+        ], 'Type', readonly=True,
+        help="Type is used to separate Tasks, Issues and "
+             "Timesheets input directly")

```

Comment:
Another option (and name could be source or origin maybe but n-t-h)
Type is based on the origin of the input (Task, Issue or Timesheet Activity Report)